lib/CMakeLists.txt

```diff
@@ -166,6 +166,68 @@ target_link_libraries(hermesvmlean_a PUBLIC
 # CMake requires at least one file for the target, so create an empty file.
 add_library(hermesvm SHARED dummy.cpp)
 add_library(hermesvmlean SHARED dummy.cpp)
+
+# Do not build a framework for the test suite.
+if(NOT HERMES_ENABLE_TEST_SUITE)
+  if(APPLE)
+    set_target_properties(hermesvm PROPERTIES OUTPUT_NAME hermes)
+
+    set_target_properties(hermesvm PROPERTIES
+      FRAMEWORK TRUE
+      VERSION ${PROJECT_VERSION}
+      SOVERSION ${PROJECT_VERSION}
+      FRAMEWORK_VERSION ${PROJECT_VERSION_MAJOR}
+      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${PROJECT_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${PROJECT_VERSION}
+      MACOSX_FRAMEWORK_IDENTIFIER dev.hermesengine.${HERMES_APPLE_TARGET_PLATFORM}
+    )
```

Comment:
This builds only for MacOS. In React Native we support also iOS, Catalyst, XR and TvOS... 😅

Reply:
Actually, these changes come from the main branch of Hermes, as shown [here](https://github.com/facebook/hermes/blob/main/API/hermes/CMakeLists.txt#L124). They were introduced in [this PR](https://github.com/facebook/hermes/pull/285), and you can see the exact changes in [this file](https://github.com/facebook/hermes/pull/332/files#diff-fd8fa405441aec2374d5056c0eac5b8548ec9a88e55545b4ff9134a069eba4eeR113). Since I'm not as familiar with the iOS build system as you are, I'm open to suggestions if you think changes need to be made 🫡